### PR TITLE
Feature/media management

### PR DESCRIPTION
**Pull request by samuelgrant**
### Mobile application API
//todo: @jacksos101 

### Media management system
//todo: @samuelgrant @jacksos101 
The media managment system contians the following features: (View #78)
- Image, audio and document uploads.
- Images can be cropped
- A gallery with a search by name and type filter
- Image gallery
- Image detials, edit and delete

**To Do:**
- Re-crop image
- Select image for media components
- Select file in link components 
- Fix checkbox in the edit detials (Show Copyright) view

### Other Changes
//todo: @samuelgrant @jacksos101

### CODE REVIEW

File: DiscoverDeepCove/Controllers/ActivitiesController.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                return Ok(_Db.Activities.Where(c => c.Active && c.Track.Active)
+                .Select(s => new
+                {
+                    s.Id,
+                    updated_at = s.UpdatedAt.ToString("yyyy-MM-dd HH:mm:ss")
+                }).ToList());
+            }
+            catch(Exception ex)
+            {
+                _Logger.LogError("Error retrieving fact file entries list: {0}", ex.Message);
```
Comment (MitchellQ) [inline DiscoverDeepCove/Controllers/ActivitiesController.cs]: Should this be `fact file entries`?

File: DiscoverDeepCove/Controllers/ActivitiesController.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                        updated_at = s.UpdatedAt.ToString("yyyy-MM-dd HH:mm:ss")
+                    }).FirstOrDefault();
+
+                if (Activity == null) return NotFound();
+
+                return Ok(Activity);
+            }
+            catch(Exception ex)
+            {
+                _Logger.LogError("Error retrieving fact file entry data: {0}", ex.Message);
```
Comment (MitchellQ) [inline DiscoverDeepCove/Controllers/ActivitiesController.cs]: Here too?

File: DiscoverDeepCove/Controllers/ActivitiesController.cs (hunk 1 above)
Comment (jacksos101) [reply DiscoverDeepCove/Controllers/ActivitiesController.cs]: Good catch, will fix

File: Resources/js/React/AdminPortal/ManageMedia.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import Gallery from './ManageMedia/Gallery';
+import Detials from './ManageMedia/Details';
+import Upload from './ManageMedia/Upload';
+
+import { Button } from '../Components/Button';
+
+
+const sections = {
+    1: Gallery,
+    2: Detials,
```
Comment (jacksos101) [inline Resources/js/React/AdminPortal/ManageMedia.js]: Typo

File: Resources/js/React/AdminPortal/ManageMedia.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
+﻿import React, { Component } from 'react';
+import { render } from 'react-dom';
+import Gallery from './ManageMedia/Gallery';
+import Detials from './ManageMedia/Details';
```
Comment (jacksos101) [inline Resources/js/React/AdminPortal/ManageMedia.js]: Typo

**Pull request merged**